promise.py

```diff
@@ -406,10 +412,52 @@ def handleSuccess(_):
                 all_promise.fulfill(values)
 
         for p in promises:
-            promisify(p).done(handleSuccess, all_promise.reject)
+            cls.promisify(p).done(handleSuccess, all_promise.reject)
 
         return all_promise
 
+    @classmethod
+    def promisify(cls, obj):
+        if isinstance(obj, Promise):
```

Comment:
In the case we create a new promise class `MyPromise` subclassing `Promise`,
I would expect the return of `MyPromise.promisify(Promise())` to be a `MyPromise` instance instead of `Promise`.

As, in the general case we are returning instances of the class (`cls()`).


Thoughts?